gitenberg/util/util.py

```diff
@@ -0,0 +1,55 @@
+import os
+from contextlib import contextmanager
+import psycopg2
+
+# an exception safer wrapper around cd
+# maybe there's a better place in the code base for this
+# from stackoverflow
+@contextmanager
+def cd(newdir):
+    prevdir = os.getcwd()
+    os.chdir(os.path.expanduser(newdir))
+    try:
+        yield
+    finally:
+        os.chdir(prevdir)
+
+def makeTables(connection, cursor, makeBook=True, makeAuthor=True, makeCover=True, makeExternalLink=True):
```

Comment:
We agreed on the new database schema for gitensite with the others on the team, so this db structure doesn't conflict with what they want.
The idea is that this will run independently of the web server, as a python process which updates the database whenever the github repos are modified.  Perhaps packaging this with gitberg isn't ideal, I'm not sure, but certainly the plan has been to have it not running on django. It isn't exactly part of the command line suite of tools that gitberg provides either. 
For now, that's why I had updateDb in a separate folder. Maybe I should put makeTables into that separate folder as well, not sure how it's best to organize it. But really makeTables is just a nice function for duplicating the database setup locally, it's not actually necessary, I just found it convenient.

Reply:
what's the reason for not using the django orm? Do you contemplate giten_site talking to this database?

Reply:
This database is planned as the database giten_site will talk to, yes. The plan has been to not use django for this just because this isn't going to be part of the webserver. I think it's a lot more straightforward (and lightweight) not to get entangled with django. I don't see any benefit of using django for this.

Reply:
if it's not part of the webserver, where is it? How is it configured?

Reply:
We discussed this previously, it's in the shared google doc weekly deliverables under data processing design.
The purpose of this code is to update the database when any git repo is updated. The webserver is connected to the database, but this process is not directly part of it. It also handles turning rdf into yaml. I will add cover generation here also.

Reply:
![weekly deliverables](https://user-images.githubusercontent.com/926513/33452599-befe7dca-d5e0-11e7-9959-42bceebd28f9.png)

So I guess we need to flesh out some things.

There are 3 yellow boxes. the one marked "python web server" has to be running continuously. is that to run in a separate server/container from the django web server running the website? If so, how will it be administered? 

The "periodically each book" process. Is that to be running somewhere else, locally perhaps? and it won't access the db, only SQS? What is it to be doing?

There is a "new book" box. Why is that treated differently from "periodically each book"?

And the big box, is it talking to two different github repos?

It looks like all the arrow point to the database, which seem to imply that we can't use it it the other direction. is that the intent?

Assuming that the django website will want to use information in the database, we'll need ORM models for the tables in the database. I'm not sure its a good idea to have db tables managed in two places.

Reply:
just to bring this current: maketables and db dependency should be removed from gitberg